LanguageFeatures/Patterns/object_A02_t01.dart

```diff
@@ -0,0 +1,69 @@
+// Copyright (c) 2022, the Dart project authors.  Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+/// @assertion objectPattern ::= typeName typeArguments? '(' patternFields? ')'
+/// ...
+/// As with record patterns, the getter name can be omitted and inferred from
+/// the variable pattern in the field subpattern which may be wrapped in a unary
+/// pattern
+///
+/// @description Checks an object pattern with omitted getter name wrapped in a
+/// different unary patterns. Test a switch expression
+/// @author sgrekhov22@gmail.com
+
+// SharedOptions=--enable-experiment=patterns,records
+
+import "../../Utils/expect.dart";
+import "patterns_lib.dart";
+
+String test(Shape shape) {
+  return switch (shape) {
+    Circle(sizeAsInt: 1, :var areaAsDouble as num)
+        => "cast = ${areaAsDouble.toStringAsFixed(2)}",
+    Circle(sizeAsInt: 2, :final areaAsNull as num)
+        => "cast = ${areaAsNull.toStringAsFixed(2)}",
+    Circle(sizeAsInt: 3, :var areaAsNullable?)
+        => "null-check = ${areaAsNullable.toStringAsFixed(2)}",
+    Circle(sizeAsInt: 4, :final areaAsNull?)
+        => "null-check = ${areaAsNull.toStringAsFixed(2)}",
+    Circle(sizeAsInt: 5, :var areaAsNullable!)
+        => "null-assert = ${areaAsNullable.toStringAsFixed(2)}",
+    Circle(sizeAsInt: 6, :var areaAsNull!)
+        => "null-assert = ${areaAsNull.toStringAsFixed(2)}",
+    Rectangle(sizeAsInt: 1, :int areaAsInt) => "variable-1 = $areaAsInt",
+    Rectangle(sizeAsInt: 2, :var areaAsInt) => "variable-2 = $areaAsInt",
+    Rectangle(sizeAsInt: 3, :final int areaAsInt) => "variable-3 = $areaAsInt",
+    Rectangle(sizeAsInt: 4, :final areaAsInt) => "variable-4 = $areaAsInt",
+    Rectangle<Meter>(sizeAsInt: 44, :final areaAsInt)
+        => "variable-5 = $areaAsInt",
+    Rectangle(sizeAsInt: 5, :(int areaAsInt)) => "parenthesized-1 = $areaAsInt",
```

Comment:
I wouldn't expect this one to be supported. We have [this](https://github.com/dart-lang/language/blob/master/accepted/future-releases/0546-patterns/feature-specification.md#:~:text=As%20with%20record%20patterns%2C%20the%20getter%20name%20can%20be%20omitted%20and%20inferred%20from%20the%20variable%20pattern%20in%20the%20field%20subpattern%20which%20may%20be%20wrapped%20in%20a%20unary%20pattern.)

> As a convenience, the identifier can be omitted and inferred from pattern. In this case the subpattern must be a variable pattern which may be wrapped in a unary pattern. 

I don't think this is worded precisely (cf. https://github.com/dart-lang/language/issues/2698).

If the language team accepts the recommended change in "'Wrapped in a unary pattern' too broad?" then this will be a compile-time error. Same for 6-8.

Reply:
Even if `parenthesizedPattern` is taken to be a unary pattern then we can't have 2, so 6-8 are errors in any case.